src/System.Linq.Expressions/src/System/Linq/Expressions/LambdaExpression.cs

```diff
@@ -113,16 +113,27 @@ public bool TailCall
         /// <returns>A delegate containing the compiled version of the lambda.</returns>
         public Delegate Compile()
         {
-#if FEATURE_CORECLR
+#if FEATURE_COMPILE
             return Compiler.LambdaCompiler.Compile(this);
 #else
             return new System.Linq.Expressions.Interpreter.LightCompiler().CompileTop(this).CreateDelegate();
 #endif 
```

Comment:
This should just forward to   Compile(preferInterpretation: true)

we cannot support case where both compiling and interpreting are unavailable anyways.